src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -25,6 +26,52 @@
 
 namespace ekat {
 
+namespace impl {
+
+  template <typename TeamMember, class Lambda, typename ValueType, bool Serialize>
```

Comment:
The standard specifies that there is no semantic difference between `class` and `typename` for (typep) template arguments (except one special case in pre C++17, but that doesn't apply here).

However, I think it _might_ be misleading for a reader (that doesn't know what I said above) to see a mix of typename and class, and he/she/they might be led to think that there are differences between the two. I would prefer to use the same keyword (either class or typename) for all the (type) template argument. At least, let's be consistent within the same `template<..>` line (consistency across different uses of `template` is less of an issue).